src/components/Details/VmDetails/VmDetails.js

```diff
@@ -50,14 +50,21 @@ export class VmDetails extends React.Component {
       k8sError: null,
       templateError: null,
       form: {},
+      preEditFormValues: {},
       template: null,
     };
   }
 
-  setEditing = editing =>
-    this.setState({
-      editing,
-    });
+  setEditing = editing => this.setState({ editing });
+
+  onCancel = () => {
+    this.setEditing(false);
```

Comment:
can we have just one `setState` call here?

Reply:
Done!